packages/collector/src/agentConnection.js

```diff
@@ -32,8 +25,24 @@ const paddingForInodeAndFileDescriptor = 200;
 const maxContentLength = 1024 * 1024 * 5;
 let maxContentErrorHasBeenLogged = false;
 
+const http = uninstrumentedHttp.http;
 let isConnected = false;
 
+/** @type {string | null} */
+let cpuSetFileContent = null;
+
+/**
+ * @param {import('@instana/core/src/util/normalizeConfig').InstanaConfig} config
+ * @param {any} _pidStore
+ */
+exports.init = function init(config, _pidStore) {
+  logger = config.logger;
+  cmdline.init(config);
+  pidStore = _pidStore;
+
+  cpuSetFileContent = getCpuSetFileContent();
```

Comment:
NOTE: Moved from root initialization to init fn. We should never initialize components on root.